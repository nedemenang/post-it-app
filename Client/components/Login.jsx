import React, {Component} from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';
import '../public/style.css';
import $ from '../public/jquery.js';
import AppActions from '../actions/AppActions';
import AppStore from '../stores/AppStore';

class Login extends Component {
  
login(event){
  // 
  event.preventDefault();
  const email = this.refs.loginEmail.value.trim();
  const password = this.refs.loginPassword.value.trim();
  
  let user = {
    email : email,
    password: password,
    isAuthenticated: false,
    profilePic: ''
  }

  if(this.refs.loginEmail.value === ''){
    AppActions.receiveErrors('Please insert email');
  }else if (this.refs.loginPassword.value === '')
  {
    AppActions.receiveErrors('Please insert password');
  } else{
    AppActions.login(user); 
  }
}

handleToggle(){
    $('form').slideToggle();
    AppActions.receiveErrors('');
    this.refs.email.value = '';
    this.refs.username.value = '';
    this.refs.password.value = '';
    this.refs.loginEmail.value = '';
    this.refs.loginPassword.value = '';
  };

signup(event){
  event.preventDefault();
  let user = {
    email : this.refs.email.value.trim(),
    password: this.refs.password.value.trim(),
    username: this.refs.username.value.trim()
  }
  

  if(this.refs.email.value === ''){
    AppActions.receiveErrors('Please insert email');
  }else if (this.refs.password.value === '')
  {
    AppActions.receiveErrors('Please insert password');
  }else if (this.refs.username.value === ''){
    AppActions.receiveErrors('Please insert username');
  } 
  else{
    AppActions.registerUser(user);
    this.refs.email.value === '';
    this.refs.username.value === '';
    this.refs.password.value === '';
  }
}

onSignIn(googleUser){
  console.log('im in on sign in method');
  const id_token = googleUser.getAuthResponse().id_token
  AppActions.registerGoogleUser(id_token);
   
}


renderGoogleLoginButton() {
    console.log('rendering google signin button')
    gapi.signin2.render('my-signin2', {
      'scope': 'https://www.googleapis.com/auth/plus.login',
      'width': 293,
      'height': 50,
      'longtitle': true,
      'theme': 'dark',
      'onsuccess': this.onSignIn
    })
  }

  componentDidMount() {
    window.addEventListener('google-loaded',this.renderGoogleLoginButton);
  }

  constructor(props){
    super(props);
    //this.state =;

    this.login = this.login.bind(this);
    this.signup = this.signup.bind(this);
    this.handleToggle = this.handleToggle.bind(this);
    this.onSignIn = this.onSignIn.bind(this);
    this.renderGoogleLoginButton = this.renderGoogleLoginButton.bind(this);
  }
  render(){
    //console.log(this.props.errors)
    return(
      <div className="login-page">
      <div className="form">
    <form className="login-form">
    <h3>LOG IN</h3>
      <input type="text" ref="loginEmail" placeholder="email"/>
      <input type="password" ref="loginPassword" placeholder="password"/>
      <p className="error">{this.props.errors}</p>
      <button className="button" onClick={this.login}>Log In</button>
            <br/>
       <br/>
      <div id="my-signin2"></div>
      <p className="message">Not registered? <a onClick={this.handleToggle} href="#">Create an account</a></p>
    </form>
    <form className="register-form">
    <h3>REGISTER</h3>
      <input type="text" ref="email" placeholder="email address"/>
      <input type="password" ref="password" placeholder="password"/>
      <input type="text" ref="username" placeholder="username"/>
      <p className="success">{this.props.success}</p>
      <p className="error">{this.props.errors}</p>
      <button className="button" onClick={this.signup}>Register</button>
      <p className="message">Already registered? <a onClick={this.handleToggle} href="#">Sign In</a></p>
    </form>
  </div>
</div>
    );
  }

}

export default Login;