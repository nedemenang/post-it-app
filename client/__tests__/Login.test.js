import React from 'react';
import mockery from 'mockery';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from '../components/Login.jsx';


// mockery.enable();
// mockery.registerMock('../public/images/logo.png', 'office smiley');
// mockery.registerMock('../public/images/8b6c67595d0f851d12e743fd3fe694ab.png',
// 'office smiley');

describe('the Login form component', () => {
  let mountedComponent;

  const str = location.href;
  const str2 = str.replace('#', ' ');

  const loginFormItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Login />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = loginFormItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = loginFormItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should call function on click of login submit button', () => {
    const component = shallow(<Login />);
    const preventDefault = jest.fn();
    component.find('#registerButton').simulate('click', { preventDefault });
    expect(preventDefault).toBeCalled();
  });

  it('should call function on click of login submit button', () => {
    const component = shallow(<Login />);
    const preventDefault = jest.fn();
    component.find('#loginButton').simulate('click', { preventDefault });
    expect(preventDefault).toBeCalled();
  });

  it('should update state on phone number field change', () => {
    loginFormItem().find('#phoneNo').simulate('change', { target: {
      value: '+23470234567'
    } });
    expect(loginFormItem().state().phoneNumber).toBe('+23470234567');
  });

  it('should update state on username field change', () => {
    loginFormItem().find('#username').simulate('change', { target: {
      value: 'newUsername'
    } });
    expect(loginFormItem().state().username).toBe('newUsername');
  });

  it('should update state on password field change', () => {
    loginFormItem().find('#password').simulate('change', { target: {
      value: 'securePassword'
    } });
    expect(loginFormItem().state().password).toBe('securePassword');
  });

  it('should update state on email field change', () => {
    loginFormItem().find('#email').simulate('change', { target: {
      value: 'unRegisteredUser@email.com'
    } });
    expect(loginFormItem().state().email).toBe('unRegisteredUser@email.com');
  });


  it('should update state on login email field change', () => {
    loginFormItem().find('#loginEmail').simulate('change', { target: {
      value: 'registeredUser@email.com'
    } });
    expect(loginFormItem().state().loginEmail).toBe('registeredUser@email.com');
  });

  it('should update state on password field change', () => {
    loginFormItem().find('#loginPassword').simulate('change', { target: {
      value: 'SecurePassword'
    } });
    expect(loginFormItem().state().loginPassword).toBe('SecurePassword');
  });

  mockery.disable();
});
