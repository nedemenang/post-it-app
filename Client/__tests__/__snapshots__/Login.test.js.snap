// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Login form component should match snapshot test 1`] = `
<div>
  <div
    className="login-image"
  >
    <img
      src={Object {}}
    />
  </div>
  <div
    className="login-page"
  >
    <div
      className="form"
    >
      <form
        className="login-form"
      >
        <h3>
          LOG IN
        </h3>
        <input
          id="loginEmail"
          onChange={[Function]}
          placeholder="email"
          type="text"
          value=""
        />
        <input
          id="loginPassword"
          onChange={[Function]}
          placeholder="password"
          type="password"
        />
        <p
          className="error"
        />
        <button
          className="button"
          id="loginButton"
          onClick={[Function]}
        >
          Log In
        </button>
        <br />
        <br />
        <div
          id="my-signin2"
        />
        <p
          className="message"
        >
          Not registered? 
          <a
            href="#"
            id="createAccount"
            onClick={[Function]}
          >
            Create an account
          </a>
        </p>
        <p
          className="message"
        >
          Forgot Password? 
          <a
            href="/PasswordReset"
          >
            Reset Password
          </a>
        </p>
      </form>
      <form
        className="register-form"
      >
        <h3>
          REGISTER
        </h3>
        <input
          id="email"
          onChange={[Function]}
          placeholder="email address"
          type="text"
          value=""
        />
        <input
          id="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
        <input
          id="username"
          onChange={[Function]}
          placeholder="username"
          type="text"
          value=""
        />
        <input
          id="phoneNo"
          onChange={[Function]}
          placeholder="Phone number (+2348012345678)"
          type="text"
          value=""
        />
        <p
          className="success"
        />
        <p
          className="error"
        />
        <button
          className="button"
          id="registerButton"
          onClick={[Function]}
        >
          Register
        </button>
        <p
          className="message"
        >
          Already registered? 
          <a
            href="#"
            onClick={[Function]}
          >
            Sign In
          </a>
        </p>
      </form>
    </div>
  </div>
</div>
`;
